packages/web-components/fast-foundation/src/slider/slider.ts

```diff
@@ -354,16 +354,17 @@ export class Slider extends FormAssociated<HTMLInputElement>
     /**
      *  Handle mouse moves during a thumb drag operation
      */
-    private handleMouseMove = (e: MouseEvent): void => {
+    private handleMouseMove = (e: MouseEvent | TouchEvent): void => {
         if (this.readOnly || this.disabled || e.defaultPrevented) {
             return;
         }
 
         // update the value based on current position
+        const sourceEvent = e instanceof TouchEvent ? e.touches[0] : e;
```

Comment:
I *think* this is going to throw a ReferenceError in Safari because `TouchEvent` is not supported. Can you test this in Safari to confirm?

Reply:
I'm not seeing any exceptions/errors in safari
![Screen Shot 2020-10-15 at 11 53 24 AM](https://user-images.githubusercontent.com/25805778/96173832-469a2300-0edd-11eb-8c5a-a25cc27bb4f8.png)

Reply:
I'm using Safari 14 though.

Reply:
It looks like Safari supports touch events in this way according to their documentation. I need to put my build somewhere public so that I can test it on my iPad though.

Reply:
I wonder if babel is aliasing that expression to `window.TouchEvent` or something: https://caniuse.com/touch claims Safari 14 does not support `TouchEvent`.

If you pop `"foo" instanceof TouchEvent` into the console of Safari you get a ReferenceError which is why I bring this up. I *do* see it referenced in [webkitjs](https://developer.apple.com/documentation/webkitjs/touchevent) documentation though so I'm a little confused. I'll build this and launch it in safari to see what the compiled source-code is

Reply:
Thanks. I dug up my old iPad mini, charged and updated it to the latest iOS. I tried setting up a public server on Azure but had login issues but realized I was over complicating it. So I set up IIS on my laptop and tried moving the storybook over but neither the one in fast-components or the one in the root directory seem to work. Is there actual files deployed to the disk or is storybook running only from memory? I tried adding an output directory to the build-storybook script in the package.json but that didn't seem to do anything at all.

Reply:
`start-storybook` is all in-memory I believe, though I could be mistaken. I also tried to find the relevant code in the generated files from `build-storybook` but couldn't, so there may be something wonky going on there.

I just confirmed it *does* throw a ReferenceError though, @marjonlynch you should see it when you start to drag the slider with the mouse.

Reply:
Assuming that this works behaviorally, you can avoid the runtime error by checking if TouchEvent exists in the window prior to checking the if the event is an instance of the TouchEvent type:

```ts
window.TouchEvent && e instanceof TouchEvent ? x : y;
```

Reply:
TypeScript won't let me. I had this same error when I tried to make it more robust with a conditional of (instanceof TouchEvent && TouchEvent.touches && TouchEvent.touches.length > 0). It treats it as if it might be an TouchEvent but one of the other cases is false. Is there a way around this in TypeScript? The only way I can think of doing it is to breaking them out into two statements like I had before and doing a check there. It's more verbose though.

Reply:
huh.... I would expect TypeScript to still restrict the type, not sure why it isn't.

I'd say in this case casting the type is okay:
```ts
const sourceEvent = window.TouchEvent && e instanceof TouchEvent ? e.touches[0] : e as MouseEvent;
```

Alternatively, I think you could test for *MouseEvent* instead:

```ts
const sourceEvent = e instanceof MouseEvent ? e : e.touches[0];
```

Both of those appease TypeScript

Reply:
That worked. I like the explicitness of the first way. Thanks.